src/main/java/modtrekt/model/task/Task.java

```diff
@@ -42,11 +52,17 @@ public boolean equals(Object other) {
         }
 
         return other != null && (other instanceof Task)
-                && ((Task) other).getDescription().equals(this.getDescription());
+                && ((Task) other).getDescription().equals(this.getDescription())
+                && ((Task) other).getModule().equals(this.getModule());
```

Comment:
Can this be simplified to use `isSameTask`? If I'm not wrong `instanceof` also checks for `null`, so the null check may be redundant.
```suggestion
                && isSameTask((Task) other);
```